alphagov/paas-cf
# [#163982078] Expose S3 buckets count metric

What
----
With the new S3 broker in private beta, we want to monitor the number of
buckets in the account compared to the account limit on buckets so that we can
request an increase preemptively.

This commit creates a new metric reader which calls the S3 API to list all
buckets, counts them, and exposes the number as a metric.

How to review
-------------
1. Code review
2. Run it down your pipeline, then check the `aws_s3_buckets_count` metric is exposed at `paas-metrics.${APPS_DNS_ZONE_NAME}/metrics` endpoint
3. Check Prometheus for a `S3BucketCountCloseToLimit` alert (shouldn't be firing)
4. Also review and merge https://github.com/alphagov/paas-aws-account-wide-terraform/pull/161 (it has been run in dev already)

Who can review
--------------
Not me
